### DISCUSSION TITLE
How to find the entry function of a program?

### DISCUSSION THREAD

**Discussion by rainsdm** [openssl/openssl#26182] [Q&A] [answered]
I want to know where the entry point of this function is.
For example, in a simple C program like this:
```c
int main(void)
{
    printf("Hello world");
    return 0;
}
```
The `main` function is its entry point. In this source code, I don't seem to find such a function or something similar. Where is its actual location? In a similar project like the Gpg source code, it's easy to find the entry point; but here, I'm having a hard time finding it.

**Comment by paulidale** [comment]
OpenSSL is a library not an application.  It doesn't have such an entry point.

**Reply by rainsdm** [reply]
But it can be compiled into an application. How does its application process the received command-line arguments?

**Reply by paulidale** [reply]
I suggest you find an introductory C programming book or guide for this.
OpenSSL isn't the place to ask such.

**Reply by mattcaswell** [reply, marked as answer]
If you're question is actually "where is the entry point for the OpenSSL command line application" - then look here:

https://github.com/openssl/openssl/blob/0f77b6f14e064120224a590139ce76ba9e027a79/apps/openssl.c#L237-L323

**Reply by rainsdm** [reply]
Thank you @mattcaswell , that's exactly what I wanted to ask.
Thank you so much.
